cxx/clean_relation.hh

```diff
@@ -392,7 +392,11 @@ class CleanRelation : public Relation<T> {
   void set_cluster_assignment_gibbs(const Domain& domain, const T_item& item,
                                     int table, std::mt19937* prng) {
     int table_current = domain.get_cluster_assignment(item);
-    assert(table != table_current);
+    if (table == table_current) {
```

Comment:
Can we use `[[maybe_unused]]` in these cases also? It'd be nice to preserve the `opt` behavior of optimizing away the assert, instead of tricking it into doing the check anyway.

Reply:
We can also compile with `-Wno-unused-variable` to suppress the warnings

Reply:
If they're optimized out either way, could we switch back to the original assert with [[maybe_unused]] on table_current (and similarly in hirm.cc) for readability?

Reply:
Done here.  I didn't change anything in hirm.cc because the [[maybe_unused]] annotation doesn't play well with the for loop bindings that are used there.